examples/custom_client.rs

```diff
@@ -0,0 +1,30 @@
+// Minimal custom client example.
+use k8s_openapi::api::core::v1::ConfigMap;
+use tower::ServiceBuilder;
+
+use kube::{
+    api::{Api, ListParams},
+    client::ConfigExt,
+    Client, Config,
+};
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    std::env::set_var("RUST_LOG", "info,kube=debug");
+    tracing_subscriber::fmt::init();
+
+    let config = Config::infer().await?;
+    let https = config.native_tls_https_connector()?;
+    let client = Client::new(
+        ServiceBuilder::new()
+            .layer(config.set_server_layer())
```

Comment:
nit: the name `set_server_layer` reads to me like it's a layer for server operation, which, while a bit nonsensical here, still makes me have to go to the docs. Maybe `base_url_layer()` (with the `set_` verb implicit) is more intuitive here?